llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp

```diff
@@ -1189,7 +1283,7 @@ bool NVPTXDAGToDAGISel::tryLoad(SDNode *N) {
   std::optional<unsigned> Opcode;
   MVT::SimpleValueType TargetVT = LD->getSimpleValueType(0).SimpleTy;
 
-  SmallVector<SDValue, 12> Ops({getI32Imm(InstructionOrdering, DL),
+  SmallVector<SDValue, 12> Ops({getI32Imm(Ordering, DL), getI32Imm(Scope, DL),
```

Comment:
We've added another element to the array. Does it bump the max number of elements ever used past 12? If so, vector size may need to be increased.

Reply:
12 is just the capacity, so the worst that may happen is that it falls back to heap allocation.
If we want a vector that fails when the static capacity is exceeded (to prevent any heap allocation fallback), using C++26 `inplace_vector<T, Capacity>` is probably a better choice. If that's the case, does LLVM have something like that?

Reply:
We can't use c++26 (I believe LLVM builds w/ c++17 at the moment).

The idea here is to make sure that the default vector size is sufficient to effectively accommodate expected uses.
Presumably previously we expected to see up to 12 elements. Now we've added another. While the array will fall back to the allocation, we do not want that. If we know we may need up to N elements, we should just say so.

There's no need to add any runtime checks, all you need to do is check the code below and see what's the largest number of additional elements we add to the array, and adjust the array size accordingly.

Reply:
I checked and we are not exceeding 12 elements here in any branch.